### ISSUE TITLE
jobwatcher does not calculate busy nodes correctly

### ISSUE THREAD

**Issue by tkuzuhara** [aws/aws-parallelcluster#941]
**Environment:**
 - AWS ParallelCluster / CfnCluster version : aws-parallelcluster-2.1.1
 - OS: Centos 7.X
 - Scheduler: SGE
 - Master instance type: N/A
 - Compute instance type: N/A

**Bug description and how to reproduce:**
The function get_busy_nodes from jobwatcher has this:
```
# get nodes reserved by running jobs 
# if a host has 1 or more job running on it, it'll be marked busy 
def get_busy_nodes(instance_properties): 
    command = "/opt/sge/bin/lx-amd64/qstat -f" 
    _output = run_command(command, {'SGE_ROOT': '/opt/sge', 
                                    'PATH': '/opt/sge/bin:/opt/sge/bin/lx-amd64:/bin:/usr/bin'}) 
    nodes = 0 
    output = _output.split("\n")[2:] 
    for line in output: 
        line_arr = line.split() 
        if len(line_arr) == 5: 
            # resv/used/tot. 
            (resv, used, total) = line_arr[2].split('/') 
            if int(used) > 0 or int(resv) > 0: 
                nodes += 1 
    return nodes 
```

It appears this code is trying to count the number of nodes from the output of `qstat -f`.  This is clearly unsafe and actually quite wrong.  Current qstat for a cluster we have deployed looks like this:
```
queuename                      qtype resv/used/tot. load_avg arch          states
---------------------------------------------------------------------------------
all.q@ip-172-31-128-100.us-wes BIP   0/4/4          17.59    lx-amd64      a
---------------------------------------------------------------------------------
all.q@ip-172-31-128-112.us-wes BIP   0/4/4          17.38    lx-amd64      a
---------------------------------------------------------------------------------
all.q@ip-172-31-128-114.us-wes BIP   0/4/4          16.82    lx-amd64      a
---------------------------------------------------------------------------------
all.q@ip-172-31-128-139.us-wes BIP   0/4/4          17.79    lx-amd64      a
---------------------------------------------------------------------------------
all.q@ip-172-31-128-14.us-west BIP   0/0/4          -NA-     lx-amd64      au
---------------------------------------------------------------------------------
all.q@ip-172-31-128-145.us-wes BIP   0/4/4          17.59    lx-amd64      a
---------------------------------------------------------------------------------
all.q@ip-172-31-128-159.us-wes BIP   0/4/4          17.62    lx-amd64      a
---------------------------------------------------------------------------------
all.q@ip-172-31-128-165.us-wes BIP   0/4/4          17.24    lx-amd64      a
```
The `a` state at the end has occurred due to an alarm status because of load thresholds.  (Note that we haven't even changed these from whatever is configured by default.)  (Second note, one of the nodes is in a 'au` state, which is a different issue because parallelcluster is not clearing the state correctly for a spot instance that was lost.)

Either way the used nodes in this state should still be counted, so looking for exactly 5 columns in a 6 column output is incorrect.  (Yes I realize it works most of the time because the states column is usually omitted on a fully functioning node.)

The problem with the above is that nodes that are running are not counted, so jobwatcher is unable to realize it needs to request more nodes when jobs are waiting.  Despite being in a soft alarm state, the nodes above are running jobs and must be counted.

**Comment by lukeseawalker** [comment]
Hi @tkuzuhara,
thank you for reporting this. I'm marking it as bug

**Comment by demartinofra** [comment]
Fixed with version 2.4.0
